client/middlewares/withAuthorization.tsx

```diff
@@ -6,23 +6,44 @@ import {
 } from "next/server";
 
 import { MiddlewareFactory } from "./types";
+import { getToken } from "next-auth/jwt";
 
 export const withAuthorization: MiddlewareFactory = (next: NextMiddleware) => {
   return async (request: NextRequest, _next: NextFetchEvent) => {
     // 👇️ vars for route management
     const { pathname } = request.nextUrl;
+    const isRouteAuth =
+      pathname.indexOf("/auth") > -1 || pathname.indexOf("/unauth") > -1;
 
-    /******************TODO***********************
-     *
-     * AUTHENTICATION
-     */
+    // 👇️ check if authentication route
+    if (isRouteAuth === true) {
+      //👌 ok: route to next middleware
+      return next(request, _next);
+    }
+    // 👇️ check for auth token: /api/auth/session
+    const token = await getToken({
+      req: request,
+      secret: process.env.NEXTAUTH_SECRET,
+    });
 
-    //...for now, route baseUrl to {onboarding)/home}
-    if (pathname.endsWith("/")) {
-      //route to (onboarding)\home
+    if (token) {
```

Comment:
I think this middleware is the source of the e2e test failure. It is preventing the test from going directly to /dashboard/operations/ (code [here](https://github.com/bcgov/cas-registration/blob/develop/client/e2e/operations.spec.ts#L20)

I think what we'll need is a method to bypass authentication in the test suite similar to what has been done in CIF / CIIP for Cypress.

I think getting this auth work in is more important than our playwright tests at the moment, so I think we should create a tech debt card for allowing playwright to bypass auth and disable e2e testing so that this work can be merged.

Reply:
Thanks, Dylan!